Patch: 70_invalid_utf8_dos.patch
Author: Lionel Elie Mamane <lionel@mamane.lu>
Avoid DOS if attachement filename is invalid Unicode string
diff -Nur Mailman/Handlers/Scrubber.py mailman-2.1.10b3/Mailman/Handlers/Scrubber.py
--- a/Mailman/Handlers/Scrubber.py	2008-02-02 23:29:26.000000000 +0100
+++ b/Mailman/Handlers/Scrubber.py	2008-02-08 23:33:16.000000000 +0100
@@ -317,7 +317,10 @@
                 os.umask(omask)
             desc = part.get('content-description', _('not available'))
             desc = Utils.oneline(desc, lcset)
-            filename = part.get_filename(_('not available'))
+            try:
+                filename = part.get_filename(_('not available'))
+            except UnicodeDecodeError:
+                filename = _('not available')
             filename = Utils.oneline(filename, lcset)
             replace_payload_by_text(part, _("""\
 A non-text attachment was scrubbed...
@@ -439,7 +442,10 @@
     ctype = msg.get_content_type()
     # i18n file name is encoded
     lcset = Utils.GetCharSet(mlist.preferred_language)
-    filename = Utils.oneline(msg.get_filename(''), lcset)
+    try:
+        filename = Utils.oneline(msg.get_filename(''), lcset)
+    except UnicodeDecodeError:
+        filename = ''
     filename, fnext = os.path.splitext(filename)
     # For safety, we should confirm this is valid ext for content-type
     # but we can use fnext if we introduce fnext filtering
